src/main/java/com/mathworks/ci/MatlabBuild.java

```diff
@@ -22,33 +22,29 @@
      * This Method decorates the launcher with MATLAB command provided and returns the Process
      * object to launch MATLAB with appropriate startup options like -r or -batch
      * 
+     * @param matlabLauncher current build launcher
      * @param workspace Current build workspace
      * @param launcher Current build launcher
      * @param listener Current build listener
-     * @param envVars Environment variables of the current build
      * @param matlabCommand MATLAB command to execute on shell
      * @return matlabLauncher returns the process launcher to run MATLAB commands
      */
-    default ProcStarter getProcessToRunMatlabCommand(FilePath workspace, Launcher launcher,
-            TaskListener listener, EnvVars envVars, String matlabCommand, String uniqueName)
+    default ProcStarter getProcessToRunMatlabCommand(ProcStarter matlabLauncher, FilePath workspace,

```

Comment:
This interface change for getProcessToRunMatlabCommand still seems a bit odd and unnecessary. The callers are creating the ProcStarter object, making a change to the cwd, and then calling getProcessToRunMatlabCommand. Instead can we
1. pass the cwd to getProcessToRunMatlabCommand as a parameter. This way the construction of the ProcStarter method is still self contained in one method. 
or
2. set the cwd on the ProcStarter object after calling getProcessToRunMatlabCommand. This way, there is no need to change the interface.

Reply:
Implemented option 2  its much feasible than option 1 . & yes i could avoid many changes with this.